packages/babel-traverse/src/path/introspection.ts

```diff
@@ -385,8 +385,11 @@ export function _guessExecutionStatusRelativeToDifferentFunctions(
   this: NodePath,
   target: NodePath,
 ): RelativeExecutionStatus {
+  const targetIsArrowExpressionDeclaration =
+    target.isArrowFunctionExpression() &&
+    t.isVariableDeclarator(target.container);
```

Comment:
What about this `const` name? Maybe we could add this validator to types?

Reply:
Let's keep this here for now, unless there is a clear need for it in multiple parts of the codebase.